File: retaguarda-software-house/node/retaguarda-sh/src/util/objeto.nfe.ts
/**
   * classe transiente que armazena o objeto enviado pelo pagseguro
   */

export class ObjetoNfe {

	cnpj: string;
	justificativa: string;
	ano: string;
	modelo: string;
	serie: string;
	numeroInicial: string;
	numeroFinal: string;
	chaveAcesso: string;
		
	/**
	* Constructor
	*/
	constructor(objetoJson: {}) {
		if (objetoJson != null) {
			this.cnpj = objetoJson['cnpj'];
			this.justificativa = objetoJson['justificativa'];
			this.ano = objetoJson['ano'];
			this.modelo = objetoJson['modelo'];
			this.serie = objetoJson['serie'];
			this.numeroInicial = objetoJson['numeroInicial'];
			this.numeroFinal = objetoJson['numeroFinal'];
			this.chaveAcesso = objetoJson['chaveAcesso'];
		}
	}
    
}

File: retaguarda-software-house/delphi/retaguarda_sh/Model/Transiente/ObjetoNfe.pas
unit ObjetoNfe;

interface

uses
  MVCFramework.Serializer.Commons, ModelBase;

type

  [MVCNameCase(ncLowerCase)]
  TObjetoNfe = class(TModelBase)
  private
    FCnpj: string;
    FJustificativa: string;
    FAno: string;
    FModelo: string;
    FSerie: string;
    FNumeroInicial: string;
    FNumeroFinal: string;
    FChaveAcesso: string;

  public
		[MVCNameAsAttribute('cnpj')]
		property Cnpj: string read FCnpj write FCnpj;
		[MVCNameAsAttribute('justificativa')]
		property Justificativa: string read FJustificativa write FJustificativa;
		[MVCNameAsAttribute('ano')]
		property Ano: string read FAno write FAno;
		[MVCNameAsAttribute('modelo')]
		property Modelo: string read FModelo write FModelo;
		[MVCNameAsAttribute('serie')]
		property Serie: string read FSerie write FSerie;
		[MVCNameAsAttribute('numeroInicial')]
		property NumeroInicial: string read FNumeroInicial write FNumeroInicial;
		[MVCNameAsAttribute('numeroFinal')]
		property NumeroFinal: string read FNumeroFinal write FNumeroFinal;
		[MVCNameAsAttribute('chaveAcesso')]
		property ChaveAcesso: string read FChaveAcesso write FChaveAcesso;

  end;

implementation

{ TEmpresa }



end.
